packages/slider/index.tsx

```diff
@@ -0,0 +1,380 @@
+  }
+
+  render() {
+    const {
+      discrete,
+      displayMarkers,
+      value,
+      min,
+      max,
+      step,
+      disabled,
+      /* eslint-disable @typescript-eslint/no-unused-vars */
+      className,
+      isRtl,
+      onValueChange,
+      onValueInput,
+      /* eslint-enable @typescript-eslint/no-unused-vars */
+      ...otherProps
+    } = this.props;
+
+    const {
+      markerValue,
+      markerStyles,
+      trackStyleProperty,
+      thumbContainerStyleProperty,
+    } = this.state;
+
+    return (
+      <div
+        className={this.classes}
+        role='slider'
+        aria-valuemin={min}
+        aria-valuemax={max}
+        aria-valuenow={value}
+        data-step={step}
+        aria-label='Select Value'
+        aria-disabled={disabled}
+        ref={this.sliderElement}
+        onBlur={this.onBlur}
+        onFocus={this.onFocus}
+        onMouseDown={this.onDown}
+        onPointerDown={this.onDown}
+        onTouchStart={this.onDown}
+        onKeyDown={this.onKeyDown}
+        {...otherProps as React.HTMLProps<HTMLDivElement>}
+      >
+        <div className='mdc-slider__track-container'>
+          <div className='mdc-slider__track' style={trackStyleProperty} />
+          {discrete && displayMarkers && (
+            <div className='mdc-slider__track-marker-container'>
+              {markerStyles.map((styles, index) => (
+                <div
+                  key={index}
+                  className='mdc-slider__track-marker'
+                  style={styles}
+                />
+              ))}
+            </div>
+          )}
+        </div>
+        <div
+          className='mdc-slider__thumb-container'
+          style={thumbContainerStyleProperty}
+          onMouseDown={this.onThumbDown}
+          onPointerDown={this.onThumbDown}
+          onTouchStart={this.onThumbDown}
+          onTransitionEnd={this.onThumbEnd}
+        >
+          {discrete && (
+            <div className='mdc-slider__pin'>
+              <span className='mdc-slider__pin-value-marker'>
+                {markerValue}
+              </span>
+            </div>
+          )}
+          <svg className='mdc-slider__thumb' width='21' height='21'>
+            <circle cx='10.5' cy='10.5' r='7.875' />
+          </svg>
+          <div className='mdc-slider__focus-ring' />
+        </div>
+      </div>
+    );
+  }
+
+  private initFoundation() {
+    if (this.foundation) {
+      this.foundation.destroy();
+    }
+    this.foundation = new MDCSliderFoundation(this.adapter);
+    this.foundation.init();
+
+    const {step, value, min, max, disabled} = this.props;
+    this.foundation.setDisabled(disabled!);
+    this.foundation.setStep(step!);
+    this.foundation.setValue(value!);
+    this.foundation.setMin(min!);
+    this.foundation.setMax(max!);
+    this.foundation.layout();
+    this.foundation.setupTrackMarker();
+
+    this.onDown = (evt) => (this.foundation as any).handleDown_(evt);
```

Comment:
Why do you need to cast as `any`?

Reply:
To eliminate the Typescript complaints in VS Code about not knowing the `private` function.

It seems the `lint` check doesn't care, though, so I can remove it... I was just trying to resolve the majority of red underlines in VS Code.